torch/csrc/utils/auto_gpu.h

```diff
@@ -29,41 +30,31 @@ struct AutoGPU {
   }
 
   ~AutoGPU() {
-#ifdef USE_CUDA
-    if (original_device != -1) {
-      cudaSetDevice(original_device);
-    }
-#endif
+    #ifdef USE_CUDA
+      // Resets the original device (if necessary)
+      if (original_device != -1) {
+        int cur_device = -1;
+        THCudaCheck(cudaGetDevice(&cur_device));
+        if (cur_device != original_device) THCudaCheck(cudaSetDevice(original_device));
+      }
```

Comment:
Is this really worth it? You'll often end up making 2 API calls instead of 1. What's the perf of `GetDevice` vs `SetDevice`

Reply:
I'm going to revert all changes to auto_gpu.h because it updated in master, anyway. So no worries here.